src/OpenFeature/Model/ResolutionDetails.cs

```diff
@@ -1,20 +1,22 @@
+using OpenFeature.Constant;
+
 namespace OpenFeature.Model
 {
     public class ResolutionDetails<T>
     {
         public T Value { get; }
         public string FlagKey { get; }
-        public string ErrorCode { get; }
+        public ErrorType ErrorType { get; }
         public string Reason { get; }
         public string Variant { get; }
 
-        public ResolutionDetails(string flagKey, T value, string errorCode = null, string reason = null, string variant = null)
+        public ResolutionDetails(string flagKey, T value, ErrorType errorType = ErrorType.None, string reason = null, string variant = null)

```

Comment:
I spent more time than I'd like to admit thinking about the parameter order here given the optionals.. I guess it doesn't matter to much, but I think this order and these defaults makes the most sense.